Sliceviewer: Unhandled exception overlaying peaks workspace on ws < 2 dimensions

### Expected behavior

Peaks viewer should be disabled for workspaces with fewer than 3 dimensions.

### Actual behavior
If I attempt to overlay a peaksws onto a workspace with fewer than 3 dimensions I receive and unhandled exception:
```
Python-[Error] Traceback (most recent call last):                                                                                
  File "/home/mgigg/Code/mantidproject/mantid/qt/python/mantidqt/widgets/sliceviewer/view.py", line 451, in peaks_overlay_clicked  
    self.presenter.overlay_peaks_workspaces()                                                                                     
  File "/home/mgigg/Code/mantidproject/mantid/qt/python/mantidqt/widgets/sliceviewer/presenter.py", line 180, in overlay_peaks_workspaces
    self._peaks_view_presenter.overlay_peaksworkspaces(names_to_overlay)                                                           
  File "/home/mgigg/Code/mantidproject/mantid/qt/python/mantidqt/widgets/sliceviewer/peaksviewer/presenter.py", line 160, in overlay_peaksworkspaces
    self.notify(PeaksViewerPresenter.Event.OverlayPeaks)                                                                         
  File "/home/mgigg/Code/mantidproject/mantid/qt/python/mantidqt/widgets/sliceviewer/peaksviewer/presenter.py", line 190, in notify
    presenter.notify(event)                                                                                                       
  File "/home/mgigg/Code/mantidproject/mantid/qt/python/mantidqt/widgets/sliceviewer/peaksviewer/presenter.py", line 64, in notify
    self._overlay_peaks()                                                                                                         
  File "/home/mgigg/Code/mantidproject/mantid/qt/python/mantidqt/widgets/sliceviewer/peaksviewer/presenter.py", line 87, in _overlay_peaks
    self.model.draw_peaks(self._view.sliceinfo, self._view.painter)                                                                
  File "/home/mgigg/Code/mantidproject/mantid/qt/python/mantidqt/widgets/sliceviewer/peaksviewer/view.py", line 52, in sliceinfo   
    return self._sliceinfo_provider.get_sliceinfo()                                                                                
  File "/home/mgigg/Code/mantidproject/mantid/qt/python/mantidqt/widgets/sliceviewer/presenter.py", line 79, in get_sliceinfo    
    range=data_view.dimensions.get_slicerange())                                                                                   
  File "/home/mgigg/Code/mantidproject/mantid/qt/python/mantidqt/widgets/sliceviewer/sliceinfo.py", line 33, in __init__          
    point), f"Expected a length 3 array for slice point, found length={len(point)}"                                           
AssertionError: Expected a length 3 array for slice point, found length=2
```

### Steps to reproduce the behavior

* Use the script in #28683 to create a matrix workspace and peaks workspace. 
* Display slice viewer on matrix workspace
* Select peaks viewer button and overlay the loaded peaks workspace
* :boom: 

### Platforms affected
All

--- a/qt/python/mantidqt/widgets/sliceviewer/model.py
+++ b/qt/python/mantidqt/widgets/sliceviewer/model.py
@@ -78,6 +78,18 @@ class SliceViewerModel:
         else:
             return False
 
+    def can_support_peaks_overlays(self) -> bool:
+        """
+        Check if the given workspace can support peaks overlay
+        """
+        try:
+            if self._get_ws().getNumDims() > 2:
+                return True
+        except AttributeError:
+            pass
+
+        return False
+
     def get_frame(self) -> SpecialCoordinateSystem:
         """Return the coordinate system of the workspace"""
         return self._ws.getSpecialCoordinateSystem()

--- a/qt/python/mantidqt/widgets/sliceviewer/peaksviewer/model.py
+++ b/qt/python/mantidqt/widgets/sliceviewer/peaksviewer/model.py
@@ -38,7 +38,6 @@ class PeaksViewerModel(TableWorkspaceDisplayModel):
     """View model for PeaksViewer
     Extends PeaksWorkspace functionality to include color selection
     """
-
     def __init__(self, peaks_ws, fg_color, bg_color):
         """
         :param peaks_ws: A pointer to the PeaksWorkspace
@@ -129,10 +128,9 @@ def create_peaksviewermodel(peaks_ws_name):
     :return: A new PeaksViewerModel object
     :raises ValueError: if the workspace referred to by the name is not a PeaksWorkspace
     """
-    return PeaksViewerModel(
-        _get_peaksworkspace(peaks_ws_name),
-        fg_color=next(FG_COLORS)['fg_color'],
-        bg_color=DEFAULT_BG_COLOR)
+    return PeaksViewerModel(_get_peaksworkspace(peaks_ws_name),
+                            fg_color=next(FG_COLORS)['fg_color'],
+                            bg_color=DEFAULT_BG_COLOR)
 
 
 # Private

--- a/qt/python/mantidqt/widgets/sliceviewer/presenter.py
+++ b/qt/python/mantidqt/widgets/sliceviewer/presenter.py
@@ -47,6 +47,8 @@ class SliceViewer(object):
         if self.model.can_normalize_workspace():
             self.view.data_view.norm_opts.currentTextChanged.connect(self.normalization_changed)
             self.view.data_view.set_normalization(ws)
+        if not self.model.can_support_peaks_overlays():
+            self.view.data_view.disable_peaks_button()
         if not self.model.can_support_nonorthogonal_axes():
             self.view.data_view.disable_nonorthogonal_axes_button()
 

--- a/qt/python/mantidqt/widgets/sliceviewer/test/test_sliceviewer_model.py
+++ b/qt/python/mantidqt/widgets/sliceviewer/test/test_sliceviewer_model.py
@@ -10,10 +10,10 @@
 import unittest
 from unittest.mock import MagicMock, patch
 
-from mantid.api import MatrixWorkspace, IMDEventWorkspace, IMDHistoWorkspace, \
-    SpecialCoordinateSystem
-from mantid.simpleapi import CreateMDHistoWorkspace, CreateWorkspace, CreateMDWorkspace, \
-    FakeMDEventData
+from mantid.api import (MatrixWorkspace, IMDEventWorkspace, IMDHistoWorkspace,
+                        SpecialCoordinateSystem)
+from mantid.simpleapi import (CreateMDHistoWorkspace, CreateWorkspace, CreateMDWorkspace,
+                              FakeMDEventData)
 from mantidqt.widgets.sliceviewer.model import SliceViewerModel, WS_TYPE
 from numpy.testing import assert_equal, assert_allclose
 import numpy as np
@@ -255,6 +255,18 @@ class SliceViewerModelTest(unittest.TestCase):
                                                   units=SpecialCoordinateSystem.QLab,
                                                   has_oriented_lattice=True)
 
+    def test_matrix_workspace_cannot_support_peaks_overlay(self):
+        self._assert_supports_peaks_overlay(False, MatrixWorkspace)
+
+    def test_md_workspace_with_fewer_than_three_dimensions_cannot_support_peaks_overlay(self):
+        for ndims in range(3):
+            self._assert_supports_peaks_overlay(False, IMDEventWorkspace, ndims=ndims)
+            self._assert_supports_peaks_overlay(False, IMDHistoWorkspace, ndims=ndims)
+
+    def test_md_workspace_with_three_or_more_dimensions_can_support_peaks_overlay(self):
+        self._assert_supports_peaks_overlay(True, IMDEventWorkspace, ndims=3)
+        self._assert_supports_peaks_overlay(True, IMDHistoWorkspace, ndims=3)
+
     def test_create_non_orthogonal_transform_raises_error_if_not_supported(self):
         model = SliceViewerModel(
             self._create_mock_workspace(MatrixWorkspace,
@@ -310,9 +322,18 @@ class SliceViewerModelTest(unittest.TestCase):
         model = SliceViewerModel(self._create_mock_workspace(ws_type, units, has_oriented_lattice))
         self.assertEqual(expectation, model.can_support_nonorthogonal_axes())
 
-    def _create_mock_workspace(self, ws_type, units, has_oriented_lattice):
+    def _assert_supports_peaks_overlay(self, expectation, ws_type, ndims=2):
+        ws = self._create_mock_workspace(ws_type,
+                                         units=SpecialCoordinateSystem.QLab,
+                                         has_oriented_lattice=False,
+                                         ndims=ndims)
+        model = SliceViewerModel(ws)
+        self.assertEqual(expectation, model.can_support_peaks_overlays())
+
+    def _create_mock_workspace(self, ws_type, units, has_oriented_lattice, ndims=2):
         ws = MagicMock(spec=ws_type)
         if hasattr(ws, 'getExperimentInfo'):
+            ws.getNumDims.return_value = ndims
             ws.getSpecialCoordinateSystem.return_value = units
             ws.getNonIntegratedDimensions.return_value = [MagicMock(), MagicMock()]
             expt_info = MagicMock()
@@ -321,6 +342,7 @@ class SliceViewerModelTest(unittest.TestCase):
             expt_info.sample.return_value = sample
             ws.getExperimentInfo.return_value = expt_info
         elif hasattr(ws, 'getNumberHistograms'):
+            ws.getNumDims.return_value = 2
             ws.getNumberHistograms.return_value = 3
             mock_dimension = MagicMock()
             mock_dimension.getNBins.return_value = 3

--- a/qt/python/mantidqt/widgets/sliceviewer/test/test_sliceviewer_presenter.py
+++ b/qt/python/mantidqt/widgets/sliceviewer/test/test_sliceviewer_presenter.py
@@ -123,6 +123,22 @@ class SliceViewerTest(unittest.TestCase):
         self.view.data_view.plot_matrix.assert_called_with(
             self.model.get_ws(), normalize=mantid.api.MDNormalization.VolumeNormalization)
 
+    def peaks_button_disabled_if_model_cannot_support_it(self):
+        self.model.get_ws_type = mock.Mock(return_value=WS_TYPE.MATRIX)
+        self.model.can_support_peaks_overlay.return_value = False
+
+        SliceViewer(None, model=self.model, view=self.view)
+
+        self.view.data_view.disable_peaks_button.assert_called_once()
+
+    def peaks_button_not_disabled_if_model_can_support_it(self):
+        self.model.get_ws_type = mock.Mock(return_value=WS_TYPE.MATRIX)
+        self.model.can_support_peaks_overlay.return_value = True
+
+        SliceViewer(None, model=self.model, view=self.view)
+
+        self.view.data_view.disable_peaks_button.assert_not_called()
+
     def test_non_orthogonal_axes_toggled_on(self):
         self.model.get_ws_type = mock.Mock(return_value=WS_TYPE.MATRIX)
         data_view_mock = self.view.data_view

